mesh.go

```diff
@@ -632,15 +636,15 @@ func batchToBuffers(shader *Shader, mesh *Mesh, mat32 glMat4, mask RGBA) {
 			// 	}
 
 			// TODO: Re enable when you have funcs like inv(), transpose() on glMat4
-		// case NormalXYZ:
-		// 	renormalizeMat := c.matrix.Inv().Transpose().gl()
-		// 	normBuf := *(destBuffs[bufIdx]).(*[]glVec3)
-		// 	for i := range mesh.normals {
-		// 		vec := renormalizeMat.Apply(mesh.normals[i])
-		// 		normBuf[i] = vec
-		// 	}
-
-			// Colors
+		case NormalXYZ:
+			// renormalizeMat := mat32.Inv().Transpose()
+			normBuf := *(destBuffs[bufIdx]).(*[]glVec3)
+			for i := range mesh.normals {
+				vec := mesh.normals[i]
+				normBuf[i] = vec
+			}
```

Comment:
It's been a while, so I'm trying to remember, but I think: In this section, I think because the mesh is being batched with the `mat32` variable applied to the verts, it needs to be applied to the normals as well. That said, I don't remember why it's the transposed inverse of the mat32.

If you look at the PositionXY swizzle, there's another optimization you can do when `if mat32 == glMat4Ident {` which is basically to copy the normal over, like you've done here.

All that said, my guess is this doesn't matter unless someone adds a phong shader, or shadows, or something like that. But I think the above code will have the wrong normals unless the mat32 is the identity matrix (which would maybe be common for 3d scenes which have large meshes.

Reply:
Yea something in the 3D example scene doesn't quite look right, like the edges of the cube aren't properly defined. I'll see if I can wrap my head around this and get it going.

Reply:
Okay, this was addressed in https://github.com/unitoftime/glitch/pull/15/commits/621c282dde030932e3d154e93c92a79432f41ace - I can confirm the spinning cube looks much better now. Before it looked like the side with light was spinning rather than the light coming in from the right. I don't know what that means in proper GL terms, but it looks more correct now 😄 

![3d](https://github.com/unitoftime/glitch/assets/10190630/3354ee40-9e6b-4869-b321-fe5776cc9f34)